org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeCoordinatorHandler.java

```diff
@@ -731,13 +738,17 @@ public ZigBeeEndpoint getEndpoint(IeeeAddress address, int endpointId) {
 
     @Override
     public void networkStateUpdated(final ZigBeeTransportState state) {
+        logger.debug("{}: networkStateUpdated called with state={}", nodeIeeeAddress, state);
         switch (state) {
             case UNINITIALISED:
                 break;
             case INITIALISING:
                 break;
             case ONLINE:
-                updateStatus(ThingStatus.ONLINE);
+                // make sure to not confuse child things
```

Comment:
As a matter of interest, why does this "confuse child things"?  Do children get the ```bridgeStatusChanged``` callback? If so, I just wonder if this transition change code shouldn't be put into the framework so that children are only notified of bridge changes?

Or am I missing something else?

Reply:
> Do children get the bridgeStatusChanged callback?

Yes I have observed this. My bridge (ember coordinator) was ONLINE and ZSS reported ONLINE again and this has triggered the `bridgeStatusChanged` of the child things.

But I think its fine for ZSS to report ONLINE if it already is ONLINE, because then the user of the ZSS library (in this case the bridgehandler) can decide what to do with this information. I.e. the bridge might have missed a transition to ONLINE before and only get the second call with ONLINE. In such a case it can decide on its own if it accepts the second ONLINE call.

WDYT?

Reply:
You are right, this check already exists:

https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManagerImpl.java#L231

So, sorry about the confusion. What I had observed was probably related to the oscillation of the bridge status, which you have fixed in ZSS now.

I have removed the check from this PR.

Reply:
Thanks.